src/Illuminate/Database/Query/Builder.php

```diff
@@ -1427,31 +1441,33 @@ public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'pag
      */
     public function getCountForPagination($columns = ['*'])
     {
-        $this->backupFieldsForCount();
+        $selectSql = $this->toSql();
```

Comment:
It will stop working if it's inside the if block, cause it needs to be the query before backing up the fields.